### ISSUE TITLE
random phase control for va1 + random phase for all unison voices

### ISSUE THREAD

**Issue by autumncheney** [stargateaudio/stargate#30]
**Briefly describe your usage of Stargate**
i just started using stargate, and i have a few feature ideas that could make this a great daw to use while also abiding under its own principles

**Describe what you want to do**
there should be a knob for random phase, more specifically how random it should be. i also think the unison voices should have their own random phases, as i noticed that they all share the same phase. random phase would make the sound much smoother

**Does your proposal align with the core project design principles?**
i don't see why not

**OPTIONAL: Describe one or more ways you propose we can accomplish this**

**Additional context**

**Comment by stargateaudio** [comment]
The current implementation runs the oscillators for a few million cycles or so, then locks their phases in and always starts from the same phase for each oscillator.  The problem this solves is inconsistent volume of notes depending on whether you happened to catch it on a low transient or a high transient.

If I understand correctly, you are asking for a mode where the oscillators run freely and do no re-sync phases on note retrigger?

**Comment by autumncheney** [comment]
no, what i mean is to set the oscillator phase to a random value every time a note is pressed. the method you described of letting oscillators run freely is one way you can achieve this. like i said, you could have a slider that determines how random the phase is, so the user wouldn't have to worry about inconsistent phases if they're only using a single oscillator
